fill in installing software and browsing the web (#37456)
mdn/content: files/en-us/learn_web_development/getting_started/environment_setup/installing_software/index.md

      <td>
        <ul>
          <li>Understand what software you need to get started.</li>
          <li>Install a code editor, some modern browsers, and a local testing server.</li>
          <li>Explore options for other common types of apps.</li>
        </ul>
      </td>
    </tr>
  </tbody>
</table>

## Code editors

A decent code editor is one of the most important things for any developer to have available on their machine. As well as being the place where you write your code, code editors have a whole host of other functionality available. We have dedicated an entire article to code editors later on in the series.

> For now, we would recommend that you install [Visual Studio Code](https://code.visualstudio.com/), as it is available across different platforms, has a great feature set and support, and is the editor we mostly use. Do this now.

Reviewer: ```suggestion
For now, we would recommend that you install [Visual Studio Code](https://code.visualstudio.com/), as it is available across different platforms, has a great feature set and support, and is the editor we mostly use. You should install this now to follow along with the rest of this article.
```

## Modern web browsers

> Having modern web browsers available is essential for web development, so you can test your websites/apps on the browsers you audience will use to access them. You also need to keep your web browsers up-to-date so that they support the latest web technologies and have the latest security fixes applied.

Reviewer: ```suggestion
Having modern web browsers available to you is essential for web development so that you can test your websites or apps on the browsers your visitors use to access them. You also need to keep your web browsers up-to-date so that they support the latest web technologies and have the latest security fixes applied.
```

  - Gecko (Android): [Mozilla Firefox](https://www.mozilla.org/en-US/firefox/browsers/mobile/android/).
  - WebKit (iOS): [Apple Safari](https://www.apple.com/safari/).
    > [!NOTE]
    > Most of the Android browsers listed above have iOS versions, but these were historically all powered by Apple's WebKit engine under the hood due to Apple's App Store rules. At the of writing, browsers are starting to create versions of their iOS browsers based on their own rendering engines, due to regulatory changes. See [Apple is finally allowing full versions of Chrome and Firefox to run on the iPhone](https://www.theverge.com/2024/1/25/24050478/apple-ios-17-4-browser-engines-eu).

## Local web servers

Normally, when you type in a web address in a browser to load a website, the files that are combined to render that site by your browser are fetched from a remote web server hosted on a server computer somewhere else in the world. You'll learn more about how this works in the next article in the series.

> When creating a website locally, you can often load up the main HTML index file directly in a browser to test it. However, some examples will need to be run through a locally-installed web server to work successfully.

Reviewer: ```suggestion
When creating a website locally (on your own machine), you can often load up the main HTML index file directly in a browser to test it. However, some examples will need to be run through a locally-installed web server to work successfully.
```

One of the easiest options we've found to make a local server available is to use a code editor extension — this way, it is available right inside your code editor. Do the following inside Visual Studio Code:

1. Open the _Extensions_ pane using the _View_ > _Extensions_ menu option.

> 2. In the "Search..." box at the top of this pane, type in "live preview". The top search result should be the _Live Preview_ extension, created by Microsoft.

Reviewer: Could also link to https://marketplace.visualstudio.com/items?itemName=ms-vscode.live-server

Author: Good idea. I've linked "_Live Preview_" to it.

## Version control tools

**Version control** tools are used by developers to manage files on servers, collaborate on a project with a team, share code and assets, and avoid editing conflicts. Right now, [Git](https://git-scm.com/) is the most popular version control system along with hosting services such as [GitHub](https://github.com/) or [GitLab](https://about.gitlab.com/).

While version control tools are essential for web development teams, you don't need to worry about them right now. We've got a module dedicated to [Version control](/en-US/docs/Learn_web_development/Core/Version_control) near the end of our Core modules series.

## Site deployment apps

After you've finished developing a website or app (on your local computer, or perhaps on a development server), you'll want to put it onto a remote web server so that your users can type in the web address associated with it, and view it on the web!

> There are various ways you can do this, from buying hosting and using an [SFTP app](/en-US/docs/Learn_web_development/Howto/Tools_and_setup/Upload_files_to_a_web_server#sftp), using a service like [GitHub Pages](https://pages.github.com/) or [Netlify](https://www.netlify.com/), or even just throwing up a quick test app to share with others using something like [Glitch](https://glitch.com/) or [CodePen](https://codepen.io/).

Reviewer: ```suggestion
There are various ways you can do this, from buying hosting and using an [SFTP app](/en-US/docs/Learn_web_development/Howto/Tools_and_setup/Upload_files_to_a_web_server#sftp), using a service like [GitHub Pages](https://pages.github.com/) or [Netlify](https://www.netlify.com/), or even setting up a quick demo to share with others using something like [Glitch](https://glitch.com/) or [CodePen](https://codepen.io/).
```
